### PR TITLE
Gentx instructions

### PR DESCRIPTION
**Pull request by anilcse**
- Add mainnet gentx instructions
- Add GA to validate gentx submission

### CODE REVIEW

File: scripts/validate-gentx.sh [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+GENTX_FILE=$(find ./$CHAIN_ID/gentxs -iname "*.json")
+LEN_GENTX=$(echo ${#GENTX_FILE})
+
+# Gentx Start date
+start="2021-03-31 15:00:00Z"
+# Compute the seconds since epoch for start date
+stTime=$(date --date="$start" +%s)
+
+# Gentx End date
+end="2021-04-06 15:00:00Z"
```
Comment (clevinson) [inline scripts/validate-gentx.sh:17]:
```suggestion
end="2021-04-09 15:00:00Z"
```

We've decided to push back the target mainnet launch to April 15th to give a full 2 weeks of candidate chain running.  I think we should update GenTx end date to April 9 (Friday), which gives a full Mon-Wed between gentx closure and launching mainnet.

File: regen-1/README.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+#### Setup Regen Ledger
+
+**Clone the repo and install regen-ledger**
+```sh
+mkdir -p $GOPATH/src/github.com/regen-network
+cd $GOPATH/src/github.com/regen-network
+git clone https://github.com/regen-network/regen-ledger && cd regen-ledger
+git fetch
+git checkout v1.0.0
```
Comment (clevinson) [inline regen-1/README.md]: Should we have people checkout v1.0.0-rc1 ? I assume `v1.0.0` final may not get tagged until sometime mid next week once we are confident in candidate chain.

File: regen-1/README.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+This section applies to the validators who wants to join the genesis.
+
+### Step-1: Initialize the chain
+```sh
+regen init --chain-id regen-1 <your_validator_moniker>
+```
+
+### Step-2: Replace the genesis
+```sh
+curl -s https://raw.githubusercontent.com/regen-network/regen-ledger/master/regen-1/prelaunch-genesis.json > $HOME/.regen/config/genesis.json
```
Comment (clevinson) [inline regen-1/README.md]:
```suggestion
curl -s https://raw.githubusercontent.com/regen-network/mainnet/main/regen-1/genesis.json > $HOME/.regen/config/genesis.json
```

Prelaunch genesis will have a distinct chain-id from mainnet genesis, so maybe we should just name hte file `genesis.json`? or `regen-1-genesis.json` ?

File: scripts/validate-gentx.sh [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    set -e
+
+    echo "GentxFile::::"
+    echo $GENTX_FILE
+
+    echo "...........Init Regen.............."
+
+    git clone https://github.com/regen-network/regen-ledger
+    cd regen-ledger
+    git checkout v1.0.0
```
Comment (clevinson) [inline scripts/validate-gentx.sh]:
```suggestion
    git checkout v1.0.0-rc1
```

File: scripts/validate-gentx.sh [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    make build
+    chmod +x ./build/regen
+
+    ./build/regen keys add $RANDOM_KEY --keyring-backend test --home $REGEN_HOME
+
+    ./build/regen init --chain-id $CHAIN_ID validator --home $REGEN_HOME
+
+    echo "..........Fetching genesis......."
+    rm -rf $REGEN_HOME/config/genesis.json
+    curl -s https://raw.githubusercontent.com/regen-network/mainnet/master/$CHAIN_ID/prelaunch-genesis.json >$REGEN_HOME/config/genesis.json
```
Comment (clevinson) [inline scripts/validate-gentx.sh]: ditto on whether this should be called `prelaunch-genesis.json`. Maybe that is actually a fine name, as it is the genesis prior to gentx submission... but if we do change it in the instructions above, we should also change it here.

File: scripts/validate-gentx.sh [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+    sed -i '/genesis_time/c\   \"genesis_time\" : \"2021-03-29T00:00:00Z\",' $REGEN_HOME/config/genesis.json
+
+    GENACC=$(cat ../$GENTX_FILE | sed -n 's|.*"delegator_address":"\([^"]*\)".*|\1|p')
+    amountquery=$(jq -r '.body.messages[0].value.amount' ../$GENTX_FILE)
```
Comment (clevinson) [inline scripts/validate-gentx.sh:70]: Is this intended to be used for anything?

We've spoken about having a max-submission requirement for GenTx. I'm waiting for final agreement on what that number is, but i'm assuming thats what this is for?

I'll let you know as soon as we have clarity on it.

**Review by clevinson — COMMENTED** [review]
Nice! Few comments below..

File: regen-1/README.md (hunk 3 above)
Comment (anilcse) [reply regen-1/README.md]: Yes, or else we can just update the chain-id, genesis time in prelaunch genesis as well

File: scripts/validate-gentx.sh (hunk 6 above)
Comment (anilcse) [reply scripts/validate-gentx.sh:70]: Yes, wanted to validate the max amount. Let me know if we have a number on this so we should update the instructions as well

File: regen-1/README.md (hunk 3 above)
Comment (clevinson) [reply regen-1/README.md]: There are a few balance updates that we will need to integrate as well, so i think we should keep them separate (prelaunch vs mainnet genesis).

File: scripts/validate-gentx.sh (hunk 1 above)
Comment (anilcse) [reply scripts/validate-gentx.sh:17]: @clevinson in the blog post it is mentioned as 6th and I am still waiting for a final call on this. So, keeping it as 6th for now and we can update later. Just want to be in-line with all other communications (announcements, blog post etc)

Comment (clevinson) [reply scripts/validate-gentx.sh:17]: Ok let's do that then!

File: scripts/validate-gentx.sh (hunk 6 above)
Comment (clevinson) [reply scripts/validate-gentx.sh:70]: Number should be 50k REGEN

**Review by clevinson — APPROVED** [review]

**Pull request merged**